### PR TITLE
Add useRates

### PR DESCRIPTION
**Pull request by FrederikBolding**
- [x] Refactor to hook
- [x] Refactor usage
- [x] Add tests
- [x] Improve fetching of non-account assets

### CODE REVIEW

**Comment by FrederikBolding** [comment]
[ch6728]

File: src/components/TransactionFlow/TxReceipt.tsx [hunk 1]
```diff
@@ -166,21 +166,21 @@ export default function TxReceipt({
...
 
-  const baseAssetRate = useCallback(() => {
+  const baseAssetRate = (() => {
     if (displayTxReceipt && path(['baseAsset'], displayTxReceipt)) {
       return getAssetRate(displayTxReceipt.baseAsset);
     } else {
       return getAssetRate(txConfig.baseAsset);
     }
-  }, [displayTxReceipt, txConfig.baseAsset]);
+  })();
```
Comment (vilsbole) [inline src/components/TransactionFlow/TxReceipt.tsx:183]: Do we really need an IIFE? Can we create a tested helper function instead?

Comment (FrederikBolding) [reply src/components/TransactionFlow/TxReceipt.tsx:183]: It's just not reusable, so I didnt feel like it made sense to abstract out?

Comment (vilsbole) [reply src/components/TransactionFlow/TxReceipt.tsx:183]: ok

**Review by vilsbole — APPROVED** [review]

**Pull request merged**